src/goto-programs/goto_program.cpp

```diff
@@ -668,6 +669,26 @@ bool goto_programt::instructiont::equals(const instructiont &other) const
   // clang-format on
 }
 
+void goto_programt::instructiont::validate(
+  const namespacet &ns,
+  const validation_modet vm) const
+{
+  validate_code_full_pick(code, ns, vm);
+  validate_expr_full_pick(guard, ns, vm);
+
+  switch(type)
+  {
+  case ASSIGN:
+    DATA_CHECK(
+      code.get_statement() == ID_assign,
+      "assign instruction should contain an assign statement");
+    DATA_CHECK(targets.empty(), "assign instruction should not have a target");
+    break;
+  default:
```

Comment:
Please explicitly enumerate all other cases here to make clear what's left to be done, and also avoid that in future some new instruction type is added without a case being inserted here.